test/index.bs

```diff
@@ -203,27 +208,38 @@ When invoked, the {{USBTest/reset()}} method MUST return a new {{Promise}}
 |promise| and run the following steps <a>in parallel</a>:
 
 1.  Let |test| be the {{USBTest}} instance on which this method was invoked.
-1.  Set |test|.{{USBTest/chosenDevice}} to <code>null</code>.
 1.  For each {{FakeUSBDevice}} |fakeDevice| previously returned by
     {{USBTest/addFakeDevice()}}, invoke
     |fakeDevice|.{{FakeUSBDevice/disconnect()}}.
 1.  <a>Resolve</a> |promise|.
 
+When invoked, the {{USBDeviceRequestEvent/respondWith(result)}} method MUST run
+the following steps <a>in parallel</a>:
+
+1.  Wait until |result| settles.
+1.  Let |device| be <code>undefined</code>.
+1.  If |result| resolved with |response| and |response| is an instance of
+    {{FakeUSBDevice}} let |device| be |response|.
+1.  Resume execution of the suspended steps defined for {{USB/requestDevice()}}.
+1.  If |device| is <code>undefined</code> behave as if the user denied the
+    permission request.
+1.  Otherwise, behave as if the user granted permission to access the device
+    represented by |device|.
+
 ## {{USB}} Behavior ## {#usb-behavior}
 
 When {{USB/requestDevice(options)}} is invoked on a {{USB}} instance
 <a>controlled by</a> a {{USBTest}} |test| the UA MUST perform the following
-steps:
+steps <a>in parallel</a>:
 
-1.  Let |filters| be a new {{FrozenArray}}.
+1.  Let |event| be a new {{USBDeviceRequestEvent}}.
+1.  Let |event|.|filters| be a new {{FrozenArray}}.
 1.  Copy the members of
     <var ignore>options</var>.{{USBDeviceRequestOptions/filters}} into
-    |filters|.
-1.  Set |test|.{{USBTest/lastFilters}} to |filters|.
-1.  <a>In parallel</a>, respond to the invocation of {{USB/requestDevice()}} as
-    if the user granted permission to access the device represented by
-    |test|.{{USBTest/chosenDevice}} or, if |test|.{{USBTest/chosenDevice}} is
-    <code>null</code>, as if the user denied the permission request.
+    |event|.|filters|.
+1.  Suspend execution of the steps defined for {{USB/requestDevice()}}.
```

Comment:
This is a little weird: you never actually say to start the https://wicg.github.io/webusb/#ref-for-dom-usb-requestdevice-5 steps, so how would you suspend them?